action/metadata-support.go

```diff
@@ -0,0 +1,178 @@
+		}
+		opt.Page = response.NextPage
+	}
+
+	// Extract a list of artifact names from the artifact data structure returned from the
+	// github API.
+	var rawArtifacts []string
+	for i := range artifacts {
+		for j := range artifacts[i].Artifacts {
+			rawArtifacts = append(rawArtifacts, *artifacts[i].Artifacts[j].Name)
+		}
+	}
+
+	// Parse raw list of artifact names into struct of Artifacts and variant names.
+	// Some builds include files that are not build artifacts; we weed those out here.
+	processedArtifacts := make(map[string][]string)
+	for i := range rawArtifacts {
+		switch {
+		case strings.Contains(rawArtifacts[i], ".json") || strings.Contains(rawArtifacts[i], ".sig") ||
+			strings.Contains(rawArtifacts[i], "_SHA256SUMS"):
+			githubactions.Infof("File %v does not match any expected pattern.", rawArtifacts[i])
+		default:
+			productName := extractProductName(rawArtifacts[i])
+			processedArtifacts[productName] = append(processedArtifacts[productName], rawArtifacts[i])
+		}
+	}
+
+	return processedArtifacts
+
+}
+
+// extractProductName accepts an artifact name, ie. "product_1.2.3-ent_amd64.deb" and
+// returns a string containing the name/version of the product, ie. "product_1.2.3-ent"
+func extractProductName(rawArtifactName string) string {
+	// Extract the product name and relevant extentions with a big ugly regex. Basically,
+	// these regexes look for the semver in their input string, then search for a delimater
+	// between the variant strings that follow semver (-dev, +ent, +ent.hsm, etc.) and the
+	// architecture/file exention. The specifics of this search vary a little for a few special
+	// cases. Finally, we return the match minus the arch/file extenion.
+	productName := ""
+	// RPMs use a . instead of a _ to delinate their extentions.
+	if strings.Contains(rawArtifactName, "rpm") {
+		re := regexp.MustCompile(`^.*(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*).*?[^\.]*`)
+		productName = re.FindString(rawArtifactName)
+		productName = correctProductNameRPM(productName)
+	} else if strings.Contains(rawArtifactName, "docker") {
+		re := regexp.MustCompile(`^.*(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*).*?[^_]*`)
+		productName = re.FindString(rawArtifactName)
+		productName = correctProductNameDocker(productName)
+	} else {
+		re := regexp.MustCompile(`^.*(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*).*?[^_]*`)
+		productName = re.FindString(rawArtifactName)
+	}
+
+	// We need to do some post-processing to make the product name we use in our data more
+	// consistent than the product names we have in our artifacts:
+	// some files use ~ instead of - in their product text, we replace here as we want product
+	// names to stick to the - format.
+	productName = strings.ReplaceAll(productName, "~", "-")
+
+	// Some artifact names have a -1 after the semver, we want to remove that for our product
+	// name.
+	if len(productName) > 1 {
+		if productName[len(productName)-2:] == "-1" {
+			productName = productName[:len(productName)-2]
+		}
+	}
```

Comment:
The strings package has function we could use here to simplify this: [strings.TrimSuffix](https://pkg.go.dev/strings#TrimSuffix)
```suggestion
        productName = strings.TrimSuffix(productName, "-1")
```